### PR TITLE
Asb api obsoletes review

### PR DESCRIPTION
**Pull request by yvesgoeleven**
- DescriptionFactory was replaced by DescriptionCustomizer on the Queues, Topics & Subscriptions extension points
- fixed the snippet for converters

### CODE REVIEW

**Comment by yvesgoeleven** [comment]
@Particular/azure-maintainers please review

File: transports/upgrades/asb-7to8.md [hunk 1]
```diff
@@ -34,6 +34,6 @@ Controlling entity creation has been greatly simplified. Instead of having to pr
 
 Due to complexity of implementation, the API to specify how the `BrokeredMessage` body is stored and retrieved by overriding the default conventions is obsoleted and the following methods were deprecated:
```
Comment (danielmarbach) [inline transports/upgrades/asb-7to8.md]: Not only the complexity but also the risks involved

File: transports/azure-service-bus/configuration/full.md [hunk 2]
```diff
@@ -61,7 +61,7 @@ The following settings are available to define how queues should be created:
  * `RequiresDuplicateDetection(bool)`: Specifies whether the queue should perform native broker duplicate detection, defaults to `false`.
  * `DuplicateDetectionHistoryTimeWindow(TimeSpan)`: The time period in which native broker duplicate detection should occur.
  * `SupportOrdering(bool)`: Best effort message ordering on the queue, defaults to `false`.
- * `DescriptionFactory(Func<string, ReadOnlySettings, QueueDescription>)`: A factory method that allows to create a `QueueDescription` object from the Azure Service Bus SDK. Use this factory method to override any (future) setting that is not supported by the Queues API.
+ * `DescriptionCustomizer(Action<QueueDescription>)`: A factory method that allows to modify a `QueueDescription` object created by the Azure Service Bus SDK. Use this factory method to override any (future) setting that is not supported by the Queues API.
```
Comment (SeanFeldman) [inline transports/azure-service-bus/configuration/full.md]: >  to override any (future) setting that is not supported by the Queues API.

 to override any (future) setting**s** that ~~is~~ **are** not supported by the Queues API.

File: transports/azure-service-bus/configuration/full.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ The following settings are available to define how topics should be created:
  * `RequiresDuplicateDetection(bool)`: Specifies whether the topic should perform native broker duplicate detection, defaults to `false`.
  * `DuplicateDetectionHistoryTimeWindow(TimeSpan)`: The time period in which native broker duplicate detection should occur.
  * `SupportOrdering(bool)`: Best effort message ordering on the topic, defaults to `false`.
- * `DescriptionFactory(Func<string, ReadOnlySettings, TopicDescription>)`: A factory method that allows to create a `TopicDescription` object from the Azure Service Bus SDK. Use this factory method to override any (future) setting that is not supported by the Topics API.
+ * `DescriptionCustomizer(Action<TopicDescription>)`: A factory method that allows to modify a `TopicDescription` object created for the Azure Service Bus SDK. Use this factory method to override any (future) setting that is not supported by the Topics API.
```
Comment (SeanFeldman) [inline transports/azure-service-bus/configuration/full.md]: > any (future) setting that is not supported by the Topics API.

Same as above, plural.

File: transports/azure-service-bus/configuration/full.md [hunk 4]
```diff
@@ -95,7 +95,7 @@ The following settings are available to define how subscriptions should be creat
  * `LockDuration(TimeSpan)`: The period of time that Azure Service Bus will lock a message before trying to redeliver it, defaults to 30 seconds.
  * `MaxDeliveryCount(int)`: Sets the maximum delivery count, defaults to the number of Immediate Retries + 1. In case Immediate Retries are disabled, the transport will default `MaxDeliveryCount` to 10 attempts.  
  * `AutoDeleteOnIdle(TimeSpan)`: Automatically deletes the subscription if it hasn't been used for the specified time period. By default the subscription will not be automatically deleted.
- * `DescriptionFactory(Func<string, string, ReadOnlySettings, SubscriptionDescription>)`: A factory method that allows to create a `SubscriptionDescription` object from the Azure Service Bus SDK. Use this factory method to override any (future) setting that is not supported by the Subscription API.
+ * `DescriptionCustomizer(Action<SubscriptionDescription>)`: A factory method that allows to modiy a `SubscriptionDescription` object created for the Azure Service Bus SDK. Use this factory method to override any (future) setting that is not supported by the Subscription API.
```
Comment (SeanFeldman) [inline transports/azure-service-bus/configuration/full.md]: > any (future) setting that is not supported by the Topics API.

Same as above, plural.

File: transports/upgrades/asb-7to8.md [hunk 5]
```diff
@@ -26,10 +26,14 @@ The bundle is set to one entity. For existing endpoints running with multiple en
 See also [Azure Service Bus Transport Topologies](/transports/azure-service-bus/topologies/).
 
 
+## Controlling entity creation
+
+Controlling entity creation has been greatly simplified. Instead of having to provide a full implementation of `DescriptionFactory` where all settings on the description object had to be provided, a customizer has been introduced `DescriptionCustomizer` where only the to be customized property can be changed. `DescriptionCustomizer` can be found on the `Queues`, `Topics` and `Subscriptions` API extension points.
```
Comment (SeanFeldman) [inline transports/upgrades/asb-7to8.md]: > greatly

"significantly"?

Not sure if this is needed 🙂

Comment (SeanFeldman) [inline transports/upgrades/asb-7to8.md]: > property can be changed

plural, "properties".

File: transports/upgrades/asb-7to8.md [hunk 6]
```diff
@@ -26,10 +26,14 @@ The bundle is set to one entity. For existing endpoints running with multiple en
...
+
 ## BrokeredMessage conventions
 
-API to specify how the `BrokeredMessage` body is stored and retrieved by overriding the default conventions is obsoleted and the following methods were deprecated:
+Due to complexity of implementation, the API to specify how the `BrokeredMessage` body is stored and retrieved by overriding the default conventions is obsoleted and the following methods were deprecated:
```
Comment (SeanFeldman) [inline transports/upgrades/asb-7to8.md]: the snippets show the following:

```c#
transport.BrokeredMessageBodyType(SupportedBrokeredMessageBodyTypes.Stream);
// OR
transport.UseBrokeredMessageToIncomingMessageConverter<CustomIncomingMessageConversion>();
```
but `BrokeredMessageBodyType` has not been deprecated.

**Pull request merged**